# Exploiting large genomic datasets for the development of a real-time loop-mediated isothermal amplification (LAMP) assay targeting SP2020 for the identification of Streptococcus pneumoniae 

This document contains the code used to analyse the data for the LAMP manuscript, with working title "Exploiting large genomic datasets for the development of a real-time loop-mediated isothermal amplification (LAMP) assay targeting SP2020 for the identification of Streptococcus pneumoniae".

## (ECCMID) Abstract
### Background
Streptococcus pneumoniae (pneumococcus) is a major global cause of morbidity and mortality. Rapid, affordable molecular diagnostics have the potential to improve the detection of pneumococci from clinical specimens. The aim of this study was to develop a real-time, loop-mediated isothermal amplification (LAMP) assay targeting the SP2020 locus, a putative GntR-family transcriptional regulator. 

### Methods
LAMP primers targeting SP2020 were designed using Prime Explorer software. The presence or absence of SP2020, and putative accuracy of SP2020 primers, were assessed in two well-curated datasets: 7,547 pneumococcal genomes and 1,825 genomes of 55 non-pneumococcal Streptococcus (NPS) species. The accuracy of the primers was investigated using an in silico PCR tool (https://bigsdb.readthedocs.io/en/latest/data_analysis/in_silico_pcr.html). A maximum of four nucleotide mismatches per primer binding region was allowed for a positive predicted in silico reaction. One set of SP2020 primers was tested in vitro on three sets of isolates: 246 pneumococci that caused invasive disease in the UK in 2022; 277 isolates of 21 different NPS species; and 71 pneumococci selected to represent unique SP2020 alleles. Real-time detection was facilitated with SYBR green and three independent dilution series were used to determine the limit of detection. 

### Results
Overall, 99.6% (7518/7547, 95% CI 99.4-99.7%) of pneumococcal genomes were positive in silico using the SP2020 LAMP primers, whereas only 0.1% (1/1825, 95% CI 0.0-0.3%) of NPS genomes were positive (Streptococcus pseudopneumoniae). The reaction time of the SP2020 LAMP in vitro assay was 12min. In vitro sensitivity was 100% (246/246, 95% CI 98.5-100%) and specificity was 100% (277/277, 95% CI 98.7-100%). All pneumococci selected to represent the 71 different SP2020 alleles tested positive in vitro using this LAMP assay. The SP2020 limit of detection was 65 copies per 25 Î¼L reaction.

### Conclusions
We utilised large genomic datasets to develop in silico primers that detected the allelic diversity of SP2020 in the pneumococcal population, and validated these primers in vitro in a highly sensitive, specific and rapid LAMP assay, which could be especially useful in low-resource settings.


